qcodes/instrument_drivers/Keysight/keysight_e4980a.py

```diff
@@ -0,0 +1,333 @@
+    )
+    CSD = MeasurementFunction(
+        "CSD", ("capacitance", "dissipation_factor"), ("F", "")
+    )
+    CSQ = MeasurementFunction(
+        'CSQ', ("capacitance", "quality_factor"), ("F", "")
+    )
+    CSRS = MeasurementFunction(
+        "CSRS", ("capacitance", "resistance"), ("F", "Ohm")
+    )
+    LPD = MeasurementFunction(
+        "LPD", ("inductance", "dissipation_factor"), ("H", "")
+    )
+    LPQ = MeasurementFunction(
+        "LPQ", ("inductance", "quality_factor"), ("H", "")
+    )
+    LPG = MeasurementFunction(
+        "LPG", ("inductance", "conductance"), ("H", "S")
+    )
+    LPRP = MeasurementFunction(
+        "LPRP", ("inductance", "resistance"), ("H", "Ohm")
+    )
+    LSD = MeasurementFunction(
+        "LSD", ("inductance", "dissipation_factor"), ("H", "")
+    )
+    LSQ = MeasurementFunction(
+        "LSQ", ("inductance", "quality_factor"), ("H", "")
+    )
+    LSRS = MeasurementFunction(
+        "LSRS", ("inductance", "resistance"), ("H", "Ohm")
+    )
+    LSRD = MeasurementFunction(
+        "LSRD", ("inductance", "resistance"), ("H", "Ohm")
+    )
+    RX = MeasurementFunction(
+        "RX", ("resistance", "reactance"), ("Ohm", "Ohm")
+    )
+    ZTD = MeasurementFunction(
+        "ZTD", ("impedance", "theta"), ("Ohm", "Degree")
+    )
+    ZTR = MeasurementFunction(
+        "ZTR", ("impedance", "theta"), ("Ohm", "Radiant")
+    )
+    GB = MeasurementFunction(
+        "GB", ("conductance", "susceptance"), ("S", "S")
+    )
+    YTD = MeasurementFunction(
+        "YTD", ("admittance", "theta"), ("Y", "Degree")
+    )
+    YTR = MeasurementFunction(
+        "YTR", ("admittance", "theta"), ("Y", "Radiant")
+    )
+    VDID = MeasurementFunction(
+        "VDID", ("voltage", "current"), ("V", "A")
+    )
+
+
+class Correction4980A(InstrumentChannel):
+    """
+    Module for correction settings.
+    """
+    def __init__(
+            self,
+            parent: VisaInstrument,
+            name: str,
+    ) -> None:
+        super().__init__(parent, name)
+
+        self.add_parameter(
+            "open",
+            set_cmd=":CORRection:OPEN",
+            docstring="Executes OPEN correction based on all frequency points."
+        )
+
+        self.add_parameter(
+            "open_state",
+            get_cmd=":CORRection:OPEN:STATe?",
+            set_cmd=":CORRection:OPEN:STATe {}",
+            val_mapping={
+                'off': 0,
+                'on': 1,
+            },
+            docstring="Enables or disable OPEN correction"
+        )
+
+        self.add_parameter(
+            "short",
+            set_cmd=":CORRection:SHORt",
+            docstring="Executes SHORT correction based on all frequency points."
+        )
+
+        self.add_parameter(
+            "short_state",
+            get_cmd=":CORRection:SHORt:STATe?",
+            set_cmd=":CORRection:SHORt:STATe {}",
+            val_mapping={
+                'off': 0,
+                'on': 1,
```

Comment:
please use create_on_off_val_mapping function :)

Reply:
Now it's using `create_on_off_val_mapping`. Thanks~